src/flp/types/fixedpoint_l2/compatible_float.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Implementations of encoding fixed point types as field elements and field elements as floats
+//! for the [`FixedPointL2BoundedVecSum`](crate::flp::types::fixedpoint_l2::FixedPointL2BoundedVecSum)
+//! type.
+
+use crate::field::{Field64, FieldElement};
+// use crate::flp::types::fixedpoint_l2::CompatibleFloat;
+use fixed::FixedI16;
+use std::fmt::Debug;
+
+/// Assign a `Float` type to this type and describe how to represent this type as an integer of the
+/// given field, and how to represent a field element as the assigned `Float` type.
+pub trait CompatibleFloat<F: FieldElement> {
+    /// The float type F can be converted into.
+    type Float: Debug + Clone;
+    /// Represent a field element as `Float`, given the number of clients `c`.
+    fn to_float(t: F, c: usize) -> Self::Float;
+    /// Represent a value of this type as an integer in the given field.
+    fn to_field_integer(t: Self) -> <F as FieldElement>::Integer;
+}
+
+impl<U> CompatibleFloat<Field64> for FixedI16<U> {
```

Comment:
yeah, this is still a bit confusing to me, as:
- if one choses any combination of field and fixed that do not fit into each other, one gets an error only upon attempting to use the functions provided by this trait, instead of being informed by the compiler that the combination is incompatible
- the associated `Float` type's precision follows from the size of the `FieldElement::Integer`, and i'm not sure i can manage to write that down in a generic fashion
- it seems the choice of field together with vector size affects the soundness error as [chris pointed out](https://github.com/divviup/libprio-rs/pull/283#discussion_r953209470), which will in turn in some way affect the differential privacy guarantee we can give IIUC. i felt like whitelisting field-fixed-float tuples might make analysis less involved and be less confusing for our end user...